lib/FlagDependencyExportsPlugin.js

```diff
@@ -15,11 +15,38 @@ const Queue = require("./util/Queue");
 /** @typedef {import("./Dependency").ExportsSpec} ExportsSpec */
 /** @typedef {import("./ExportsInfo")} ExportsInfo */
 /** @typedef {import("./Module")} Module */
+/** @typedef {import("./ModuleGraph")} ModuleGraph */
+/** @typedef {import("../declarations/WebpackOptions").WebpackOptionsNormalized} WebpackOptions */
 
 const PLUGIN_NAME = "FlagDependencyExportsPlugin";
 const PLUGIN_LOGGER_NAME = `webpack.${PLUGIN_NAME}`;
 
 class FlagDependencyExportsPlugin {
+	/**
+	 * @param {WebpackOptions} webpackOptions webpack options
+	 */
+	constructor(webpackOptions) {
+		this._webpackOptions = webpackOptions;
+	}
+
+	/**
+	 *
+	 * @param {Module} module module
+	 * @param {ModuleGraph} moduleGraph module graph
+	 * @returns {boolean} need flag exportsInfo
+	 */
+	needFlagModuleExports(module, moduleGraph) {
+		const { optimization, output } = this._webpackOptions;
+		const { providedExports } = optimization;
+		const { library } = output;
+		if (providedExports === false && library && library.type === "module") {
+			// only flag the entry module
+			let connection = Array.from(moduleGraph.getIncomingConnections(module));
```

Comment:
Is there any other better way to determine whether it is an entry module during `finishModules` stage?